data/mods/ssb/moves.ts

```diff
@@ -511,6 +511,61 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Flying",
 	},
 
+	// Coconut
+	devolutionbeam: {
+		accuracy: 100,
+		basePower: 80,
+		category: "Special",
+		desc: "If the target Pokemon is evolved, this move will reduce the target to its first-stage form. If the target Pokemon is single-stage or is already in its first-stage form, this move deals damage and forces the target out. Hits Ghost types.",
+		shortDesc: "Devolves evolved mons; dmg+forces small mons out.",
+		name: "Devolution Beam",
+		pp: 5,
+		priority: 0,
+		flags: {protect: 1},
+		ignoreImmunity: {'Normal': true},
+		onTryMovePriority: 100,
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Psywave', target);
+		},
+		onModifyMove(move, source, target) {
+			let species = target.species;
+			if (species.isMega) species = this.dex.getSpecies(species.baseSpecies);
+			const isSingleStage = (species.nfe && !species.prevo) || (!species.nfe && !species.prevo);
+			if (!isSingleStage) {
+				move.category = 'Status';
+			}
+		},
+		onBasePower(basePower, source, target) {
+			let species = target.species;
+			if (species.isMega) species = this.dex.getSpecies(species.baseSpecies);
+			const isSingleStage = (species.nfe && !species.prevo) || (!species.nfe && !species.prevo);
+			if (!isSingleStage) return 0;
+		},
+		onHit(target, source, move) {
+			let species = target.species;
+			if (species.isMega) species = this.dex.getSpecies(species.baseSpecies);
+			const ability = target.ability;
+			const isSingleStage = (species.nfe && !species.prevo) || (!species.nfe && !species.prevo);
+			if (isSingleStage) {
+				target.forceSwitchFlag = true;
+			} else {
+				let prevo = species.prevo;
+				if (this.dex.getSpecies(prevo).prevo) {
+					prevo = this.dex.getSpecies(prevo).prevo;
+				}
```

Comment:
I feel like this is just going one cycle. Say blaziken will just become combusken

Reply:
wait nvm im blind